pkg/networkaware/networkoverhead/networkoverhead.go

```diff
@@ -0,0 +1,578 @@
+	agName := networkawareutil.GetPodAppGroupLabel(pod)
+	if len(agName) == 0 { // Return
+		return nil
+	}
+
+	// Get AppGroup CR
+	appGroup, err := no.findAppGroupNetworkOverhead(agName)
+	if err != nil {
+		klog.ErrorS(err, "Error while returning AppGroup")
+		return nil
+	}
+
+	// Get NetworkTopology CR
+	networkTopology, err := no.findNetworkTopologyNetworkOverhead()
+	if err != nil {
+		klog.ErrorS(err, "Error while returning NetworkTopology")
+		return nil
+	}
+
+	klog.V(6).InfoS("AppGroup CR: ", appGroup.Name)
+	klog.V(6).InfoS("Network Topology CR: ", networkTopology.Name)
+
+	// Get Dependencies of the given pod
+	dependencyList := networkawareutil.GetDependencyList(pod, appGroup)
+
+	// If the pod has no dependencies, return
+	if dependencyList == nil {
+		return nil
+	}
+
+	// Get pods from lister
+	selector := labels.Set(map[string]string{agv1alpha1.AppGroupLabel: agName}).AsSelector()
+	pods, err := no.podLister.List(selector)
+	if err != nil {
+		klog.ErrorS(err, "Error while returning pods from appGroup, return")
+		return nil
+	}
+
+	// Return if pods are not yet allocated for the AppGroup...
+	if pods == nil {
+		klog.ErrorS(err, "No pods yet allocated, return")
+		return nil
+	}
+
+	// Pods already scheduled: Get Scheduled List (Deployment name, replicaID, hostname)
+	scheduledList := networkawareutil.GetScheduledList(pods)
+	klog.V(6).InfoS("scheduledList: ", scheduledList)
+
+	// Check if scheduledList is empty...
+	if scheduledList == nil {
+		klog.ErrorS(err, "Scheduled list is empty, return")
+		return nil
+	}
+
+	// Check if bandwidth and network requirements can be met
+	region := networkawareutil.GetNodeRegion(nodeInfo.Node())
+	zone := networkawareutil.GetNodeZone(nodeInfo.Node())
+	klog.V(6).InfoS("Node Region: ", region)
+	klog.V(6).InfoS("Node Zone: ", zone)
+
+	// Sort Costs if manual weights were selected
+	no.sortNetworkTopologyCosts(networkTopology)
+
+	// Create map for cost / destinations. Search for requirements faster...
+	var costMap = make(map[networkawareutil.CostKey]int64)
+
+	// Populate Costmap
+	no.populateCostMap(costMap, networkTopology, region, zone)
+	klog.V(6).InfoS("costMap: ", costMap)
+
+	numOK, numNotOK, ok := checkMaxNetworkCostRequirements(scheduledList, dependencyList, nodeInfo, region, zone, costMap, no)
+	if ok != nil {
+		return framework.NewStatus(framework.Error, "pod hostname not found")
+	}
+
+	klog.V(4).InfoS("NumNotOk: %v / numOK: %v ", numNotOK, numOK)
+
+	if numNotOK > numOK {
+		return framework.NewStatus(framework.Unschedulable,
+			fmt.Sprintf("Node %v does not meet several network requirements from Workload dependencies: OK: %v NotOK: %v", nodeInfo.Node().Name, numOK, numNotOK))
+	}
+	return nil
+}
+
+// Score : evaluate score for a node
+func (no *NetworkOverhead) Score(ctx context.Context, cycleState *framework.CycleState, pod *v1.Pod, nodeName string) (int64, *framework.Status) {
+
+	klog.V(4).InfoS("Calculating score for pod %q on node %q", pod.GetName(), nodeName)
```

Comment:
Ditto. It's of little value to log for every Score(), maybe remove it or increase the verbose level.